src/features/recovery/services/recovery-sender.ts

```diff
@@ -28,7 +29,9 @@ async function getDelayModifierContract({
   const isSmartContract = await isSmartContractWallet(wallet.chainId, wallet.address)
 
   // Use unchecked signer for smart contract wallets as transactions do not necessarily immediately execute
-  const signer = await provider.getSigner()
+  const signer = isSmartContract
+    ? new UncheckedJsonRpcSigner(provider, await (await provider.getSigner()).getAddress())
+    : await provider.getSigner()
```

Comment:
If we need to call `await provider.getSigner()` in both cases I suggest to extract it from the shorthand condition.